### DISCUSSION TITLE
Unable to connect to mqtt: ECONNREFUSED

### DISCUSSION THREAD

**Discussion by roja45** [tsightler/ring-mqtt#388] [General Discussion]
I am entering the mqtt url as below (I can connect to the mqtt instance using an external application):
mqtt://hassos:MY_PASSWORD@IP ADDRESS OF MQTT SERVER:1883

but I get the following error:
2022-06-07T17:03:48.557Z ring-mqtt Unable to connect to MQTT broker getaddrinfo ENOTFOUND hassos

I am entering the mqtt url as below (I can connect to the mqtt instance using an external application):
mqtt://hassos:<password>@<ip address>:1883

but I get the following error:
2022-06-07T17:03:48.557Z ring-mqtt Unable to connect to MQTT broker getaddrinfo ENOTFOUND hassos

the connection info in the logs is not correct, and different to the url I have entered.

2022-06-07T17:03:38.993Z ring-mqtt Configuration file: /data/options.json
2022-06-07T17:03:38.997Z ring-mqtt Configured MQTT Host: hassos
2022-06-07T17:03:38.999Z ring-mqtt Discovered MQTT Port: 1883
2022-06-07T17:03:39.000Z ring-mqtt Configured MQTT User: 
2022-06-07T17:03:39.001Z ring-mqtt MQTT URL: **mqtt://hassos:1883**

Thanks for your help.

**Comment by tsightler** [comment]
> mqtt://hassos:@:1883

Well, this is completely invalid URL so yes, that would never work.  Can you please post the MQTT URL you have configured (obfuscate the password if you like) and also, if you upgraded from 4.x, please make sure you have removed any old configurations options as noted in the Upgrading to 5.x documentation.

Also, verify that your password has no characters that would be considered invalid in a URL.  You can still have those characters if you want, but you'll need to enter them in URL encoded format.

**Reply by roja45** [reply]
URL I have added to the configuration is the one in the second line above.  I use this same url to connect using MQTT Explorer.

Encoding the password did the trick, it had a # in it.

Thanks

**Reply by tsightler** [reply]
I'll add a note to the docs about special characters needing to be URL encoded.  Glad that worked!
